modules/topics/server/models/topic.server.model.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+/**
+ * Module dependencies.
+ */
+var mongoose = require('mongoose'),
+  Schema = mongoose.Schema,
+  crypto = require('crypto'),
+  validator = require('validator');
+
+var TopicSchema = new Schema({
+  name: {
+    type: String,
+    unique: 'Topic name already exists',
+    required: 'Topic must have a name',
+    minlength: [1, 'Topic name must contain at least one character'],
+    maxlength: [20, 'Topic name exceeds the maximum allowed length ({MAXLENGTH}).'],
+    trim: true // Removes surrounding whitespace
+  },
+  creator: {
+    type: Schema.Types.ObjectId,
+    ref: 'User'
+  },
+  updated: {
+    type: Date,
+    default: Date.now
+  },
+  created: {
+    type: Date,
+    default: Date.now
+  },
+  words: {
+    type: [String],
+    required: true
+  },
+  popularity: {
+    type: Number,
+    default: 0
+  }
```

Comment:
Not sure what I want to use popularity for yet. Maybe number of uses, or maybe add some kind of vote functionality later.